### PR TITLE
Parse XML with namespace

### LINKED ISSUE #132: Parse XML with namespace
Hello,
First, thanks for this promising tool set. And I hope I send the question on the good canal and repository. 

I try to map an xml with namespaces for nodes (an `xsd` type file).
When I remove the namespaces from my source file, the test triples are generated. 
But when I restaure namespace in the xml file and add xsd: ns to xpath, I get an empty set of triples. 

As I find no example of "xml with namespace" parsing, I ask myself how I can do that.

Here is the example I try to tackle, this can be added to mattey.
Thanks for you help, 
regards
* source xml file with ns
```xml
<?xml version="1.0" encoding="utf-8"?>
<xsd:schema xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:hropen="https://hropenstandards.org/schema/xml/" targetNamespace="https://hropenstandards.org/schema/xml/" version="4.3.0" id="PersonType" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <xsd:include schemaLocation="PersonPhysicalInclusion.xsd" />
  <xsd:include schemaLocation="PersonLegalInclusion.xsd" />
  <xsd:include schemaLocation="../profile/PersonProfileInclusion.xsd" />
  <xsd:include schemaLocation="PersonBaseType.xsd" />
  <xsd:complexType name="PersonType">
    <xsd:annotation>
      <xsd:documentation>A schema that represents all of the information of a person.</xsd:documentation>
    </xsd:annotation>
    <xsd:all>
      <xsd:element minOccurs="0" maxOccurs="1" name="legalId" type="hropen:IdentifierType">
        <xsd:annotation>
          <xsd:documentation>The legal identifier of a person. The issuer is most likely a country or state.</xsd:documentation>
        </xsd:annotation>
      </xsd:element>
    </xsd:all>
  </xsd:complexType>
</xsd:schema>
```
* yarrrml configuration
```yaml
prefixes:
  rml: 'http://semweb.mmlab.be/ns/rml#'
  rr: 'http://www.w3.org/ns/r2rml#'
  ql: 'http://semweb.mmlab.be/ns/ql#'
  rdf: 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'
  "": 'http://example.org/rules/'
  schema: 'http://schema.org/'
  dbo: 'http://dbpedia.org/ontology/'
  test: 'https://test.fr/mm'
  fno: 'https://w3id.org/function/ontology#'
  fnoi: 'https://w3id.org/function/vocabulary/implementation#'
  fnom: 'https://w3id.org/function/vocabulary/mapping#'
  fnml: 'http://semweb.mmlab.be/ns/fnml#'
  grel: 'http://users.ugent.be/~bjdmeest/function/grel.ttl#'
  xsd: 'http://www.w3.org/2001/XMLSchema'
mappings:
  mapping0:
    sources:
      - [Supergirl.xml~xpath, /xsd:schema/xsd:complexType/xsd:all/xsd:element]
    s: 'http://example.org/character/$(@name)'
    po:
      - [a, 'schema:PersonTEST~iri']
      - ['schema:TARGET', $(@type)]
      - ['schema:DESCR', $(xsd:annotation/xsd:documentation)]
      - ['dbo:SHACL', $(@minOccurs)]
```

### PR DESCRIPTION
**Pull request by florent-andre**
feat: add xml-ns xpath compatiblity
closes #132
﻿- chore: set-up the basic test and changes
- fix: test case
- fix: xpath for namespaced strings
- update buildnumber

### CODE REVIEW

File: buildNumber.properties [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #maven.buildNumber.plugin properties file
```
Comment (DylanVanAssche) [inline buildNumber.properties:1]: Can you drop this commit? We only dump these things when releasing.

File: src/main/java/be/ugent/rml/records/XMLRecord.java [hunk 2]
```diff
@@ -17,9 +21,11 @@
 public class XMLRecord extends Record {
 
     private Node node;
+    private Document document;
 
-    public XMLRecord(Node node) {
+    public XMLRecord(Node node,Document document) {
```
Comment (DylanVanAssche) [inline src/main/java/be/ugent/rml/records/XMLRecord.java:26]: Nitpick: space between args.

File: src/main/java/be/ugent/rml/records/XMLRecordFactory.java [hunk 3]
```diff
@@ -34,10 +37,10 @@
 
         try {
             XPath xPath = XPathFactory.newInstance().newXPath();
+            xPath.setNamespaceContext(new NamespaceResolver(document));
             NodeList result = (NodeList) xPath.compile(iterator).evaluate(document, XPathConstants.NODESET);
-
             for (int i = 0; i < result.getLength(); i ++) {
-                records.add(new XMLRecord(result.item(i)));
+                records.add(new XMLRecord(result.item(i),document));
```
Comment (DylanVanAssche) [inline src/main/java/be/ugent/rml/records/XMLRecordFactory.java:43]: Nitpick: space between args.

File: src/main/java/be/ugent/rml/records/xpath/NamespaceResolver.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.Iterator;
+
+import javax.xml.XMLConstants;
+import javax.xml.namespace.NamespaceContext;
+
+import org.w3c.dom.Document;
+
+// source: https://howtodoinjava.com/java/xml/xpath-namespace-resolution-example/
+
+public class NamespaceResolver implements NamespaceContext{
```
Comment (DylanVanAssche) [inline src/main/java/be/ugent/rml/records/xpath/NamespaceResolver.java:12]: Nitpick: space missing between `NamespaceContext` and `{`

File: src/test/java/be/ugent/rml/Mapper_XML_Test.java [hunk 5]
```diff
@@ -202,4 +202,9 @@ public void evaluate_1006_XML() {
     public void evaluate_1011_XML() {
         doMapping("./test-cases/RMLTC1011-XML/mapping.ttl", "./test-cases/RMLTC1011-XML/output.nq");
     }
+
+    @Test
+    public void evaluate___NS___XML() {
```
Comment (DylanVanAssche) [inline src/test/java/be/ugent/rml/Mapper_XML_Test.java:207]: You can name it `RMLTC1026-XML`, same for the directory of the test case files.

File: src/test/resources/test-cases/RMLTC__NS__-XML/student-partially-NS.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+
+<!-- This is a `partially namespaced tree`. Name,Address,... use default NS -->
```
Comment (DylanVanAssche) [inline src/test/resources/test-cases/RMLTC__NS__-XML/student-partially-NS.xml:3]: You mentioned that the partially namespaced tree wasn't investigated yet.
The files are here, could you try to add the RML mapping rules so we have a full test case as well?
Maybe it is not hard to support partially namespaced tree as well?

File: buildNumber.properties (hunk 1 above)
Comment (florent-andre) [reply buildNumber.properties]: revert done

**Comment by DylanVanAssche** [comment]
Your code is now merged into v4.13.0!
I will close the PR here.

**Pull request closed without merging**